azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/sync_clients.py

```diff
@@ -89,43 +73,61 @@ def from_authentication_provider(cls, authentication_provider, transport_name):
             raise ValueError("No specific transport can be instantiated based on the choice.")
         return cls(transport)
 
+
+class GenericClientSync(GenericClient):
+    """A superclass representing a generic synchronous client. This class needs to be extended for specific clients.
+    """
+
+    def __init__(self, transport):
+        """Initializer for a generic synchronous client.
+
+        This initializer should not be called directly.
+        Instead, the class method `from_authentication_provider` should be used to create a client object.
+
+        :param transport: The transport that the client will use.
+        """
+        super(GenericClientSync, self).__init__(transport)
+        self._inbox_manager = InboxManager(inbox_type=SyncClientInbox)
+
     def connect(self):
-        """Connects the client to an Azure IoT Hub or Azure IoT Edge instance.
+        """Connects the client to an Azure IoT Hub or Azure IoT Edge Hub instance.
 
         The destination is chosen based on the credentials passed via the auth_provider parameter
         that was provided when this object was initialized.
 
         This is a synchronous call, meaning that this function will not return until the connection
         to the service has been completely established.
         """
-        logger.info("connecting to transport")
+        logger.info("Connecting to Hub...")
 
         connect_complete = Event()
 
         def callback():
             connect_complete.set()
+            logger.info("Successfully connected to Hub")
 
-        self._transport.connect(callback)
+        self._transport.connect(callback=callback)
```

Comment:
>callback [](start = 32, length = 8)

semantics: I've been wondering if we should call this completion_callback to communicate intention.  I don't want to change this now -- I'm just curious to hear if you have any ideas.

Reply:
I agree, it is a completion callback, and I wouldn't be opposed to the added clarity

---
In reply to: [264338304](https://github.com/Azure/azure-iot-sdk-python-preview/pull/55#discussion_r264338304) [](ancestors = 264338304)